filemonitor/main.cpp

```diff
@@ -1,43 +1,20 @@
+#include "FileMonitor.hpp"
+#include <memory>
 #include <iostream>
-#include <fstream>
-#include <filesystem>
-#include <chrono>
-#include <thread>
 
-namespace fs = std::filesystem;
-
-int main() {
-    // define path to your file
-    fs::path filePath = "../test.txt";
-
-    // get last modification time
-    auto lastModified = fs::last_write_time(filePath);
-
-    while (true) {
-        // wait till next modification time check
-        std::this_thread::sleep_for(std::chrono::milliseconds(500)); 
+int main(int argc, char* argv[]) {
+    if (argc != 2) {
+        std::cerr << "Usage: " << argv[0] << " <filename>" << std::endl;
+        return 1;
+    }
 
-        // get last modification time
-        auto currentModified = fs::last_write_time(filePath);
+    std::string filePath = argv[1];
 
-        // compare two modification time, if different it's mean file was modified
-        if (currentModified != lastModified) {
-            std::cout << "file was modified!" << std::endl;
-            lastModified = currentModified;
+    // Utwórz obiekt ConcreteFileMonitor
+    std::unique_ptr<IFileMonitor> monitor = std::make_unique<FileMonitor>(filePath);
```

Comment:
"const auto monitor = std::make_unique<...>" would be much more readable

Reply:
updated